### PR TITLE
Use Vazir font with a node module instead of CDN

### LINKED ISSUE #27541: Remove dependency of Fonts CDNs in the PrestaShop open source project
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

Yes, this is related to a problem, as described below.

It appears that Google Fonts service has been judged as not compatible with GDPR and then should not be used in the PrestaShop open source service. TLDR: Google Fonts is a CDN, and it captures the IP of the website visitors (merchants, customers) before any consent or approval.

Legal documents:
- Original in German at [https://rewis.io/urteile/urteil/lhm-20-01-2022-3-o-1749320/](https://rewis.io/urteile/urteil/lhm-20-01-2022-3-o-1749320/)
- [Translation in English](https://rewis-io.translate.goog/urteile/urteil/lhm-20-01-2022-3-o-1749320/?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=en-US&_x_tr_pto=op,wapp) (sorry for using Google Translation, too long for Deepl...)

### Describe the solution you'd like

Google Fonts must be removed from the entire PrestaShop project (BO, and FO theme), and instead, a set of default fonts must be provided inside the installer so fonts are downloaded directly on the website's server.

I can provide assistance to select fonts that have a compatible license with OSL so it is possible to distribute them with the installer. (work to be done with PM and designers)

### Alternatives you've considered

Find a way to get the consent of end users before the website is actually displayed, but how?
(so, during installation, the first step would be to ask the user if they agree with Google Fonts, not wonderful for adoption. And for shop customers, this should be managed in the GDPR banner)

And, what happened if the end user disapprove?
Serving the default font in the operating system or browser? Can this generate display problems if the defauts fonts are not similar?

No easy solution. The best way to be sure to be legal compliant is to remove Google Font and include a set of open source fonts.

### Additional context

Some third party themes might also rely on Google Fonts, but this is not our problem: vendors will have to inform their customers/vendors, and provide a validation system regarding GDPR.

Also, we must show and promote best practices in term of legal compliance with the PrestaShop open source project.

**See the discussion, there are other CDNs used in the project instead of embedding fonts in the project. They must be removed too.**

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Using a CDN exposes users to a risk of data abuse and privacy abuse, so we swap to a local font using npm
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #27541.
| How to test?      | RTL language font should be loaded normally (tested it, works fine), also when you run `npm run build` on default, new-theme and classic/_dev folder, it should generate a thirdPartyNotice.json file in the asset folder, containing every notices
| Possible impacts? | RTL font

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27611)

### CODE REVIEW

**Comment by NeOMakinG** [comment]
Can be QA by a dev I think so!

**Review by matks — DISMISSED** [review]
I remember we discussed using a CDN, yes or no, for another topic.

Maybe we should add a mention in devdocs FAQ about this?

**Comment by NeOMakinG** [comment]
> I remember we discussed using a CDN, yes or no, for another topic.
> 
> Maybe we should add a mention in devdocs FAQ about this?

could probably be useful yes

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by mparvazi** [comment]
Great improvement :100: 

Some suggestions:

1. Just need to use `woff` and `woff2` for font-face. Why you should be responsible for customers who are stuck in the past? (IE or Opera Mini for BO :neutral_face: ). Even Microsoft has been planned to get rid of these type of users.
```
@font-face {
  font-family: Vazir;
  font-style: normal;
  font-display: swap;
  font-weight: 300;
  src: url("~vazir-font/dist/Vazir-Light.woff2") format("woff2"), url("~vazir-font/dist/Vazir-Light.woff") format("woff");
}
@font-face {
  font-family: Vazir;
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url("~vazir-font/dist/Vazir-Regular.woff2") format("woff2"), url("~vazir-font/dist/Vazir-Regular.woff") format("woff");
}
@font-face {
  font-family: Vazir;
  font-style: normal;
  font-display: swap;
  font-weight: 700;
  src: url("~vazir-font/dist/Vazir-Bold.woff2") format("woff2"), url("~vazir-font/dist/Vazir-Bold.woff") format("woff");
}
```
_Fonts will be reduced from 15 to 6 in public folder._

2. The `rtl_rtl.css` will be generated by CSSJanus in public folder, but only `rtl.css` will be used, so need to prevent transforming `rtl.scss` by CSSJanus. Because in future you may put overrides of RTL inside `rtl.scss` (plan to stop using theme.rtlfix).
`src/Core/Localization/RTL/StylesheetGenerator.php`
proccess

3. We don't have any call for `rtl.css` in Login page, so the `font-face` doesn't use in page. I think need to add extra call for Login page in `AdminController.php` but I'm not sure where!
Screenshot from 2022-02-26 15-55-55

**Comment by mparvazi** [comment]
Great improvement :100: 

Some suggestions:

1- Just need to use `woff` and `woff2` for font-face (to me `woff2` is enough!)
Why you should be responsible for users who are stuck in the past? (`eot` only uses by IE, `ttf` only needed for very old browsers). IMO a browser that doesn't support `woff2`, also it doesn't support some components in BO.
```
@font-face {
  font-family: Vazir;
  font-style: normal;
  font-display: swap;
  font-weight: 300;
  src: url("~vazir-font/dist/Vazir-Light.woff2") format("woff2"), url("~vazir-font/dist/Vazir-Light.woff") format("woff");
}

@font-face {
  font-family: Vazir;
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url("~vazir-font/dist/Vazir-Regular.woff2") format("woff2"), url("~vazir-font/dist/Vazir-Regular.woff") format("woff");
}

@font-face {
  font-family: Vazir;
  font-style: normal;
  font-display: swap;
  font-weight: 700;
  src: url("~vazir-font/dist/Vazir-Bold.woff2") format("woff2"), url("~vazir-font/dist/Vazir-Bold.woff") format("woff");
}
```
_The number of fonts will be reduced from 15 to 6 in the public folder._ 

2- The `rtl_rtl.css` will be generated by CSSJanus in public folder, but only `rtl.css` will be used, so need to prevent transforming `rtl.css` by CSSJanus (PHP plugin that use by `StylesheetGenerator.php`). Because in future you may put overrides of RTL inside `rtl.scss` (plan to stop using theme.rtlfix).
`src/Core/Localization/RTL/StylesheetGenerator.php`
proccess

3- We don't have any call for `rtl.css` in Login page:
`controllers/admin/AdminLoginController.php`
Screenshot from 2022-02-27 11-41-24

4- The license text for [vazir-font](https://github.com/rastikerdar/vazir-font) is null. I think it's related to missing "LICENSE" file in project's root folder.
Screenshot from 2022-02-26 16-21-05
Can use `replenishDefaultLicenseTexts` option (when this is enabled, default license texts are taken from `spdx.org` for packages where no license text was found):
Screenshot from 2022-02-27 11-43-34

5- After webpack build for default theme, there is an empty `rtl.bundle.js` file, so we can use [webpack-remove-empty-scripts](https://github.com/webdiscus/webpack-remove-empty-scripts).

**Comment by NeOMakinG** [comment]
@mparvazi Every points are fixed, except the rtl.css with CSSJanus, actually we can't do it, it's not a problem for the moment but we need to take care about the fact that no one add something else than fonts inside rtl.scss

**Comment by mparvazi** [comment]
> @mparvazi Every points are fixed, except the rtl.css with CSSJanus, actually we can't do it, it's not a problem for the moment but we need to take care about the fact that no one add something else than fonts inside rtl.scss

I'm the one who want to add override rules for RTL inside `rtl.scss`  :smile: 
Then there is no need for `theme.rtlfix` for BO theme.

**Review by PierreRambaud — DISMISSED** [review]

**Comment by mparvazi** [comment]
@NeOMakinG 
I tested and works fine.

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
@Robin-Fischer-PS @NeOMakinG I suggest a dev does the QA?

**Comment by NeOMakinG** [comment]
@matks yeah, why not!

**Review by Progi1984 — APPROVED** [review]

**Comment by Robin-Fischer-PS** [comment]
@matks @NeOMakinG I need some more info : Is there a high risk of regressions on RTL display ? If so, I would like to have a regression test made by one of the QA girls.

**Comment by NeOMakinG** [comment]
> @matks @NeOMakinG I need some more info: Is there a high risk of regressions on RTL display? If so, I would like to have a regression test made by one of the QA girls.

- RTL won't work fine anymore on IE11 (which is not supported anymore in june) and some very old browsers with low usage
- No high risk of regressions on other browsers

**Comment by Robin-Fischer-PS** [comment]
OK for QA by dev then ;)

Thanks for the quick answer @NeOMakinG :) (and sorry for my late answer, I forgot to respond yesterday :man_facepalming: )

**Comment by mparvazi** [comment]
@NeOMakinG 

The `Vazir` font renamed to `Vazirmatn` and its structure has been changed (font going to be added to the Google fonts).

https://github.com/rastikerdar/vazirmatn

Please suspend merging of this PR, package structure will be checked by Google again in couple of next days.
I'll be contact with author to make sure structure is stable and ready to use.

**Comment by NeOMakinG** [comment]
converted to draft as @mparvazi warned about something, let's wait

**Comment by mparvazi** [comment]
@NeOMakinG 

Fonts added to Google fonts:
`https://github.com/google/fonts/pull/4352`

So it's ready to use.

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Needs a rebase and Shouldn't be a draft.

**Comment by mparvazi** [comment]
@NeOMakinG 

Ready to use:
https://github.com/rastikerdar/vazirmatn

**Comment by NeOMakinG** [comment]
> Needs a rebase and Shouldn't be a draft.

The draft mode was needed because we were blocking the PR, vazir has been updated to a new package.

I removed classic stuff and changed the vazir package, it's ready for review now @mparvazi and @PrestaShop/prestashop-maintainers

File: admin-dev/themes/new-theme/package.json
Comment (mparvazi) [inline admin-dev/themes/new-theme/package.json]:
```suggestion
    "vazirmatn": "^32.1.0",
```

File: admin-dev/themes/default/package.json
Comment (mparvazi) [inline admin-dev/themes/default/package.json]:
```suggestion
    "vazirmatn": "^32.1.0"
```

File: admin-dev/themes/default/webpack.config.js
Comment (mparvazi) [inline admin-dev/themes/default/webpack.config.js]:
```suggestion
          'vazirmatn@32.1.0': 'OFL-1.1'
```

File: admin-dev/themes/new-theme/.webpack/common.js
Comment (mparvazi) [inline admin-dev/themes/new-theme/.webpack/common.js]:
```suggestion
        'vazirmatn@32.1.0': 'OFL-1.1',
```

**Comment by mparvazi** [comment]
There is a problem in v32.0.0 about kerning in numbers.
That issue is related to different displaying in windows 10 and Linux.

Fixed in v32.1

File: admin-dev/themes/new-theme/scss/rtl.scss
Comment (Progi1984) [inline admin-dev/themes/new-theme/scss/rtl.scss]:
```suggestion

```

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Comment by matks** [comment]
I checked out the branch and tested. As far as I see it works as expected.

**Pull request merged**
